src/Common/tests/System/Net/Http/Http2LoopbackServer.cs

```diff
@@ -168,6 +168,14 @@ public override async Task<HttpRequestData> HandleRequestAsync(HttpStatusCode st
             return requestData;
         }
 
+        public override async Task AcceptConnectionAsync(Func<GenericLoopbackConnection, Task> funcAsync)
+        {
+            using (Http2LoopbackConnection connection = await EstablishConnectionAsync().ConfigureAwait(false))
```

Comment:
won't this immediately dispose connection after accepting? (since we added Dispose it will now immediately shutdown, is that expected?)

Reply:
This simply implements old behavior. Remaining it will make test cleanup more difficult. Having said that, I don't really have strong feeling about naming and I'm wrong person to argue about english.

Reply:
fair enough, this kind of structures are often found in functional languages and are often named with something starting with `With` to make it clear that the lifetime of the object is scoped to the body of the lambda